### DISCUSSION TITLE
The CognitoPreTokenGenerationV2Event does not adhere to the AWS documentation

### DISCUSSION THREAD

**Discussion by DizzyDeveloper** [aws/aws-lambda-dotnet#1792] [Q&A] [answered]
According to the the AWS documentation for the [Pre token generation Lambda trigger](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-pre-token-generation.html) the claims response of the CognitoPreTokenGenerationV2Event should be able to accept any one of [String, Number, Boolean, Array of strings, numbers, booleans, or a combination of any of these, JSON]. However, at the moment it just accepts a string map?
Is there any intention implement the documented specs?

Note, I did a hack around locally by cloning the `CognitoPreTokenGenerationV2Event` and updating the signature of  ClaimsToAddOrOverride  property in the  AccessTokenGeneration class from:

`public Dictionary<string, string> ClaimsToAddOrOverride { get; set; } = new ();`
to 
`public Dictionary<string, object> ClaimsToAddOrOverride { get; set; } = new ();`

and that works. Can something similar be done as part of this repo.

**Comment by ashishdhingra** [comment, marked as answer]
Based on discussion with the team, converting this discussion into bug. Please track progress there at https://github.com/aws/aws-lambda-dotnet/issues/1798.
